uaclient/data_types.py

```diff
@@ -50,6 +61,22 @@ def from_value(val: Any) -> Any:
         return val
 
 
+class EnumDataValue(DataValue):
+    """
+    To be used for parsing enum values
+    from_value raises an error if the value is not in the enum class values
+    and returns the value if found.
+    """
+
+    @classmethod
+    def from_value(cls, val: Any) -> str:
```

Comment:
We'll need to do something similar to DataObject for the typing here. Something like:
```python
E = TypeVar("E", bound="EnumDataValue")
class EnumDataValue(DataValue, Enum):
    @classmethod
    def from_value(cls: Type[E], val: Any) -> E:
        try:
            return cls(val)
        except ValueError:
            values = [i.value for i in cls]
            raise IncorrectEnumValueError(values, cls)
```